spec/tictactoe_spec.rb

```diff
@@ -50,69 +50,116 @@ def is_row_outputed(io:, row:)
-      expectedBoardOutput = [' | | ', '------', 'X| | ', '------', ' | | ']
+      expectedBoardOutput = [" | | \n------\nX| | \n------\n | | \n"]
 
       expect(fakeIO.stdout).to eql expectedBoardOutput
     end
   end
 
   context 'when the user interacts with the board' do
-    let(:desired_position){  $desired_position =  4  }
+    let(:desired_position) { $desired_position =  4  }
 
     context 'when the user tries to place a token in a free space' do
       it 'prompt the user to select a space' do
-        fakeIO = FakeIO.new( stdin: desired_position )
-        game = TicTacToe.new( io: fakeIO )
+        fakeIO = FakeIO.new(stdin: desired_position)
+        game = TicTacToe.new(io: fakeIO)
 
         game.begin_player_turn
-        user_is_prompted_to_select_space  = fakeIO.stdout.grep(/[S|s]elect.*position/).any?
+        user_is_prompted_to_select_space = fakeIO.stdout.grep(/select.*position/i).any?
 
-        expect( user_is_prompted_to_select_space ).to eql true
+        expect(user_is_prompted_to_select_space).to eql true
       end
 
       it 'adds their token to the board' do
-        fakeIO = FakeIO.new( stdin: desired_position )
+        fakeIO = FakeIO.new(stdin: desired_position)
         player_token = 'X'
         game = TicTacToe.new(io: fakeIO)
 
         updated_board = game.begin_player_turn
 
-        is_row_outputed(io: fakeIO, row: " |#{player_token}| ")
-        expect( updated_board[desired_position] ).to eql player_token
+        is_row_outputed(io: fakeIO, row:/ \|X\| /)
+        expect(updated_board[desired_position]).to eql player_token
       end
     end
 
     context 'when the user tries to place a token in an occupied space' do
       it "should prompt the user that it's taken" do
-        fakeIO = FakeIO.new( stdin: desired_position )
+        fakeIO = FakeIO.new(stdin: desired_position)
+        game = TicTacToe.new(io: fakeIO)
+
+        game.begin_player_turn
+
+        fakeIO.stdin = desired_position
+        game.begin_player_turn
+
+        user_is_prompted_position_taken = fakeIO.stdout.grep(/position.*taken/i).any?
+        expect(user_is_prompted_position_taken).to eql true
+      end
+      it 'should prompt the user to choose another position' do
+        fakeIO = FakeIO.new(stdin: desired_position)
         game = TicTacToe.new(io: fakeIO)
 
         game.begin_player_turn
 
         fakeIO.stdin = desired_position
         game.begin_player_turn
 
-        user_is_prompted_position_taken = fakeIO.stdout.grep(/[P|p]osition.*taken/).any? 
-        expect( user_is_prompted_position_taken  ).to eql true
+        user_is_prompted_to_select_space = fakeIO.stdout.grep(/select another position/i).any?
+        expect(user_is_prompted_to_select_space).to eql true
+      end
+    end
+
+    context 'when the user input is invalid' do
+      context 'if the input is outside of the specified range' do
+        it 'prompt the user to choose another position' do
+          fakeIO = FakeIO.new(stdin: -1)
```

Comment:
Yes

Reply:
I'll convert it to a string since the IO is always a string. It wouldn't make sense to input an integer. I also realized it doesn't implicitly convert it, so that was my bad.